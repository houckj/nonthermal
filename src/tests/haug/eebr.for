C     Electron-electron bremsstrahlung, diff. cross section lab system
      DIMENSION S1(2),S2(2)
      REAL*8 A1,A2,ALPHA,ARC,AR0,CT,CT0,DTH,E1,E11,EN1,F1,F2,G0,G1,G2,
     1 G3,G4,G5,H1,H2,H3,H4,H5,H6,H7,H8,K,KMAX,KQ,L,L1,L2,L3,L4,MCQ,
     2 P1,P1Q,PEN,PI,PM,R1,R2,R44,RK,RO,RQ,RQ2,RQ4,RX1,RX2,RX3,S,S1,
     3 S2,TH,THO,THETA,W,W2,W4,W44,WIQ,WIQC,WQ,WQ2,WQ2Q,WQ4,WQ4Q,WR,
     4 WR1,WW,WWQ,X,X1,X1Q,X2,X2Q,X3,X3Q,X12,XQ,XR,ZPIA
      DATA ALPHA/0.7297353D-2/,AR0/0.5794675D-3/,MCQ/0.5109989D3/,
     1 PI/0.31415926536D1/,EN1/0.3D3/,PEN/0.1D3/,DTH/0.1D1/,NTH/180/
      NTH1=NTH+1
      ARC=PI/1.80D2
      ZPIA=0.2D1*PI*ALPHA
      WRITE(10,10) EN1
   10 FORMAT(' Electron-electron bremsstrahlung in the lab system for E1
     1 =',F6.1,' keV'/)
      E11=EN1/MCQ
      E1=E11+0.1D1
      P1Q=E11*(E11+0.2D1)
      P1=DSQRT(P1Q)
      WQ2=E1+E1
      WQ2Q=WQ2*WQ2
      WQ=WQ2+0.2D1
      W=DSQRT(WQ)
      WQ4=WQ2-0.2D1
      WQ4Q=WQ4*WQ4
      W44=0.25D0*WQ4
      WW=DSQRT(WQ4)
      WWQ=WQ*WQ4
      A1=ZPIA*WQ2/(W*WW)
      F1=(DEXP(A1)-0.1D1)/A1
      KMAX=E11/(E1-P1+0.1D1)
      PM=KMAX*MCQ
      WRITE(10,11) PM
   11 FORMAT(' Maximum photon energy =',F7.2,' keV'/)
      K=PEN/MCQ
      IF(K.GT.KMAX) GO TO 60
      KQ=K*K
      RK=0.1D1/K
      THO=180.D0
      CT0=(E1+0.1D1-E11/K)/P1
      IF(CT0.LT.-0.1D1) GO TO 15
      THO=DACOS(CT0)/ARC
   15 WRITE(10,16) PEN,THO
   16 FORMAT(' Photon energy =',F6.1,' keV, theta0 =',F6.1,' degrees:'/)
      THETA=0.D0
      DO 50 J=1,NTH1
      IF(THETA.GT.THO) GO TO 50
      TH=THETA*ARC
      CT=DCOS(TH)
      M=1
      X1=K*(E1-P1*CT)
      X2=K
      X1Q=X1*X1
      RX1=0.1D1/X1
      X3=X1
      X3Q=X1Q
      RX3=RX1
      X2Q=KQ
      RX2=RK
      X12=X1*X2
      X=X1+X2
      XQ=X*X
      H1=(X1-X2)/X
      RQ=WQ-X-X
      RO=DSQRT(RQ)
      RQ2=RQ-0.2D1
      RQ4=RQ-0.4D1
      R44=0.25D0*RQ4
      WR=DSQRT(RQ4)
      A2=ZPIA*RQ2/(RO*WR)
      F2=(DEXP(A2)-0.1D1)/A2
      XR=X12/RQ
      L1=DLOG(0.5D0*(RO+WR))
      W4=WW*DSQRT(W44*RQ4+0.4D1*XR)
      H2=W*WR+RO*WW
      L3=DLOG(0.125D0*H2*H2/X)
      L4=DLOG(0.1D1+(0.25D0*WR*W4+0.2D1*W44*R44)/XR)
      H3=0.125D0*RQ2/X*(RQ4+WQ)**2
      H4=0.4D1+0.8D1*WQ2/WQ4Q+0.5D0*RQ*RQ2/X+0.25D0*X/X12*(WQ2Q
     1+RQ2*RQ4-0.8D1*RQ2/WQ4)
      H5=0.2D1/WQ4Q*(WQ*WQ2*RQ4-RQ2-RQ2+0.4D1*RQ/WQ)
      H6=0.1D1+0.125D0*RQ2/X12*(WQ2Q+RQ2*RQ2-0.6D1*(WQ4+RQ)
     1+0.16D2*X/WQ4)+0.2D1/X12+RQ4/WQ4-0.8D1/WQ4Q
      H7=0.25D0*(RQ+WQ)/X12*H1*H1-0.25D0*(RX1-RX2)**2-0.5D0*RQ/XQ
     1+0.2D1/(WQ4Q*XR)*(0.1D1+WQ4)
      H8=RQ/WWQ*(0.12D2*WQ2Q/WQ4*X12-XQ*(0.8D1+0.48D2/WWQ))
   20 R1=RQ4+0.4D1*(X1+X1Q/RQ)
      WR1=DSQRT(R1)
      R2=RQ4+X1+X1
      W2=DSQRT(R44*X2Q+0.2D1*X*XR)
      L=DLOG(0.25D0*RO*RX1*(R2+WR*WR1))
      L2=DLOG(0.1D1+RQ*RX1/X*(R44*X2+0.5D0*WR*W2))
      G0=H7+H8/(X1Q*X1Q)-RQ*X/(W44*X1Q*X1)+RQ/X1Q*(0.4D1/WQ-0.15D1
     1-(0.8D1/WQ4-WQ2*X)/WWQ)-RQ*RX1/WQ4+((WQ-0.4D1*X2)/RQ
     2-0.25D0*WWQ/X12-0.4D1*RX1+(WQ-0.5D0*WQ2*RQ)/X1Q)/R1
      G1=0.4D1+(0.9D1-E1)*RX2-0.2D1*RX1*(WQ+WQ-X2+0.4D1)+0.75D0*WWQ/X12
     1+0.4D1*RX1/R2*(X2-0.3D1*X1+0.4D1+(RQ+RQ-0.6D1)*RX1)+RX1/R1
     2*(WQ4-0.4D1*XR)*(0.25D0*WQ*R2*RX2-RQ2-X1-X1)
      G2=RO*((RQ+0.2D1)/XQ+0.8D1/X1Q)
      G3=0.2D1*WQ2*RX2-(RQ2-X2)*RX1+0.5D0*RQ2*(RQ+X1)/X+WQ4/R2-0.2D1
     1+H3*RX1+RX1/R2*(RQ2*((WQ2-X)*X2-WQ2)-X2Q-X2Q-0.4D1*X2)+(RQ2*
     2(0.15D1*RQ4-0.5D0*WQ*(RQ-0.5D1))+X1Q+X1Q-0.6D1*X1+WQ2*X2)/(R2*X)
      G4=H4-0.15D1*WQ2*RX2+RX1*(X2Q-WQ-WQ-WQ*X2+X2+0.5D0*WQ2Q)
     1-0.5D0*X2*WQ2/X-0.25D0*RX2*(WQ4+RQ)/X*WQ2Q+0.4D1*XQ*RX1/WWQ
     2-(0.2D1*RQ2*X2+WQ4*X-0.4D1*WQ2+0.5D0*RX1*(0.8D1-RQ)*WQ2)/R2
     3+(0.5D0*RQ2*(0.3D1*RQ4-WQ*(RQ-0.5D1))-X1*(WQ-X1-X1+0.4D1))/(X*R2)
     4+H5*RX1+WQ2*X/(W44*X1Q)*(0.12D2*X/WWQ+X-E1)*(0.1D1-X*RX1/WQ)
      G5=H6-(0.2D1+X1+X1)*RX2
      S=WR*G0+G1*L/WR1+G2*L1+G3*L2/W2+0.2D1*G4*RO*L3/(W*WW*X1)
     1-G5*L4/W4
      S1(M)=S
      S2(M)=S/F2
      IF(M.EQ.2) GO TO 30
      M=2
      X1=X2
      X1Q=X2Q
      RX1=RX2
      X2=X3
      X2Q=X3Q
      RX2=RX3
      GO TO 20 
   30 WIQ=AR0*K*(S1(1)+S1(2))/(PI*RO*W*WW*MCQ)
      WIQC=AR0*K*F1*(S2(1)+S2(2))/(PI*RO*W*WW*MCQ)
      WRITE(10,40) THETA,WIQ,WIQC
   40 FORMAT(' theta =',F6.1,': CS =',E12.5,' b/(keV*sr), CSC =',
     1 E12.5,' b/(keV*sr)')
   50 THETA=THETA+DTH
      STOP
   60 WRITE(6,70)
   70 FORMAT(' The chosen photon energy is higher than the maximum photo
     1n energy')
      STOP  
      END 
