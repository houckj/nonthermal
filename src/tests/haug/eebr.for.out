 Electron-electron bremsstrahlung in the lab system for E1 = 300.0 keV

 Maximum photon energy = 221.46 keV

 Photon energy = 100.0 keV, theta0 = 109.6 degrees:

 theta =   0.0: CS = 0.20361E-04 b/(keV*sr), CSC = 0.20248E-04 b/(keV*sr)
 theta =   1.0: CS = 0.20352E-04 b/(keV*sr), CSC = 0.20238E-04 b/(keV*sr)
 theta =   2.0: CS = 0.20324E-04 b/(keV*sr), CSC = 0.20211E-04 b/(keV*sr)
 theta =   3.0: CS = 0.20277E-04 b/(keV*sr), CSC = 0.20164E-04 b/(keV*sr)
 theta =   4.0: CS = 0.20209E-04 b/(keV*sr), CSC = 0.20096E-04 b/(keV*sr)
 theta =   5.0: CS = 0.20119E-04 b/(keV*sr), CSC = 0.20006E-04 b/(keV*sr)
 theta =   6.0: CS = 0.20005E-04 b/(keV*sr), CSC = 0.19893E-04 b/(keV*sr)
 theta =   7.0: CS = 0.19865E-04 b/(keV*sr), CSC = 0.19753E-04 b/(keV*sr)
 theta =   8.0: CS = 0.19697E-04 b/(keV*sr), CSC = 0.19585E-04 b/(keV*sr)
 theta =   9.0: CS = 0.19498E-04 b/(keV*sr), CSC = 0.19388E-04 b/(keV*sr)
 theta =  10.0: CS = 0.19268E-04 b/(keV*sr), CSC = 0.19158E-04 b/(keV*sr)
 theta =  11.0: CS = 0.19005E-04 b/(keV*sr), CSC = 0.18896E-04 b/(keV*sr)
 theta =  12.0: CS = 0.18708E-04 b/(keV*sr), CSC = 0.18600E-04 b/(keV*sr)
 theta =  13.0: CS = 0.18376E-04 b/(keV*sr), CSC = 0.18270E-04 b/(keV*sr)
 theta =  14.0: CS = 0.18011E-04 b/(keV*sr), CSC = 0.17906E-04 b/(keV*sr)
 theta =  15.0: CS = 0.17611E-04 b/(keV*sr), CSC = 0.17508E-04 b/(keV*sr)
 theta =  16.0: CS = 0.17180E-04 b/(keV*sr), CSC = 0.17078E-04 b/(keV*sr)
 theta =  17.0: CS = 0.16718E-04 b/(keV*sr), CSC = 0.16618E-04 b/(keV*sr)
 theta =  18.0: CS = 0.16228E-04 b/(keV*sr), CSC = 0.16130E-04 b/(keV*sr)
 theta =  19.0: CS = 0.15712E-04 b/(keV*sr), CSC = 0.15617E-04 b/(keV*sr)
 theta =  20.0: CS = 0.15174E-04 b/(keV*sr), CSC = 0.15081E-04 b/(keV*sr)
 theta =  21.0: CS = 0.14617E-04 b/(keV*sr), CSC = 0.14527E-04 b/(keV*sr)
 theta =  22.0: CS = 0.14044E-04 b/(keV*sr), CSC = 0.13957E-04 b/(keV*sr)
 theta =  23.0: CS = 0.13459E-04 b/(keV*sr), CSC = 0.13375E-04 b/(keV*sr)
 theta =  24.0: CS = 0.12866E-04 b/(keV*sr), CSC = 0.12784E-04 b/(keV*sr)
 theta =  25.0: CS = 0.12268E-04 b/(keV*sr), CSC = 0.12189E-04 b/(keV*sr)
 theta =  26.0: CS = 0.11669E-04 b/(keV*sr), CSC = 0.11593E-04 b/(keV*sr)
 theta =  27.0: CS = 0.11072E-04 b/(keV*sr), CSC = 0.10999E-04 b/(keV*sr)
 theta =  28.0: CS = 0.10479E-04 b/(keV*sr), CSC = 0.10410E-04 b/(keV*sr)
 theta =  29.0: CS = 0.98950E-05 b/(keV*sr), CSC = 0.98283E-05 b/(keV*sr)
 theta =  30.0: CS = 0.93213E-05 b/(keV*sr), CSC = 0.92577E-05 b/(keV*sr)
 theta =  31.0: CS = 0.87606E-05 b/(keV*sr), CSC = 0.86999E-05 b/(keV*sr)
 theta =  32.0: CS = 0.82148E-05 b/(keV*sr), CSC = 0.81571E-05 b/(keV*sr)
 theta =  33.0: CS = 0.76857E-05 b/(keV*sr), CSC = 0.76310E-05 b/(keV*sr)
 theta =  34.0: CS = 0.71750E-05 b/(keV*sr), CSC = 0.71232E-05 b/(keV*sr)
 theta =  35.0: CS = 0.66838E-05 b/(keV*sr), CSC = 0.66348E-05 b/(keV*sr)
 theta =  36.0: CS = 0.62131E-05 b/(keV*sr), CSC = 0.61668E-05 b/(keV*sr)
 theta =  37.0: CS = 0.57636E-05 b/(keV*sr), CSC = 0.57200E-05 b/(keV*sr)
 theta =  38.0: CS = 0.53358E-05 b/(keV*sr), CSC = 0.52948E-05 b/(keV*sr)
 theta =  39.0: CS = 0.49301E-05 b/(keV*sr), CSC = 0.48916E-05 b/(keV*sr)
 theta =  40.0: CS = 0.45466E-05 b/(keV*sr), CSC = 0.45105E-05 b/(keV*sr)
 theta =  41.0: CS = 0.41853E-05 b/(keV*sr), CSC = 0.41515E-05 b/(keV*sr)
 theta =  42.0: CS = 0.38459E-05 b/(keV*sr), CSC = 0.38143E-05 b/(keV*sr)
 theta =  43.0: CS = 0.35281E-05 b/(keV*sr), CSC = 0.34986E-05 b/(keV*sr)
 theta =  44.0: CS = 0.32316E-05 b/(keV*sr), CSC = 0.32041E-05 b/(keV*sr)
 theta =  45.0: CS = 0.29558E-05 b/(keV*sr), CSC = 0.29302E-05 b/(keV*sr)
 theta =  46.0: CS = 0.27002E-05 b/(keV*sr), CSC = 0.26763E-05 b/(keV*sr)
 theta =  47.0: CS = 0.24639E-05 b/(keV*sr), CSC = 0.24417E-05 b/(keV*sr)
 theta =  48.0: CS = 0.22464E-05 b/(keV*sr), CSC = 0.22258E-05 b/(keV*sr)
 theta =  49.0: CS = 0.20469E-05 b/(keV*sr), CSC = 0.20277E-05 b/(keV*sr)
 theta =  50.0: CS = 0.18645E-05 b/(keV*sr), CSC = 0.18467E-05 b/(keV*sr)
 theta =  51.0: CS = 0.16986E-05 b/(keV*sr), CSC = 0.16820E-05 b/(keV*sr)
 theta =  52.0: CS = 0.15482E-05 b/(keV*sr), CSC = 0.15327E-05 b/(keV*sr)
 theta =  53.0: CS = 0.14125E-05 b/(keV*sr), CSC = 0.13981E-05 b/(keV*sr)
 theta =  54.0: CS = 0.12907E-05 b/(keV*sr), CSC = 0.12773E-05 b/(keV*sr)
 theta =  55.0: CS = 0.11820E-05 b/(keV*sr), CSC = 0.11694E-05 b/(keV*sr)
 theta =  56.0: CS = 0.10855E-05 b/(keV*sr), CSC = 0.10737E-05 b/(keV*sr)
 theta =  57.0: CS = 0.10005E-05 b/(keV*sr), CSC = 0.98935E-06 b/(keV*sr)
 theta =  58.0: CS = 0.92610E-06 b/(keV*sr), CSC = 0.91557E-06 b/(keV*sr)
 theta =  59.0: CS = 0.86162E-06 b/(keV*sr), CSC = 0.85160E-06 b/(keV*sr)
 theta =  60.0: CS = 0.80629E-06 b/(keV*sr), CSC = 0.79669E-06 b/(keV*sr)
 theta =  61.0: CS = 0.75938E-06 b/(keV*sr), CSC = 0.75012E-06 b/(keV*sr)
 theta =  62.0: CS = 0.72019E-06 b/(keV*sr), CSC = 0.71119E-06 b/(keV*sr)
 theta =  63.0: CS = 0.68804E-06 b/(keV*sr), CSC = 0.67922E-06 b/(keV*sr)
 theta =  64.0: CS = 0.66227E-06 b/(keV*sr), CSC = 0.65357E-06 b/(keV*sr)
 theta =  65.0: CS = 0.64228E-06 b/(keV*sr), CSC = 0.63362E-06 b/(keV*sr)
 theta =  66.0: CS = 0.62745E-06 b/(keV*sr), CSC = 0.61877E-06 b/(keV*sr)
 theta =  67.0: CS = 0.61722E-06 b/(keV*sr), CSC = 0.60845E-06 b/(keV*sr)
 theta =  68.0: CS = 0.61104E-06 b/(keV*sr), CSC = 0.60213E-06 b/(keV*sr)
 theta =  69.0: CS = 0.60841E-06 b/(keV*sr), CSC = 0.59929E-06 b/(keV*sr)
 theta =  70.0: CS = 0.60882E-06 b/(keV*sr), CSC = 0.59943E-06 b/(keV*sr)
 theta =  71.0: CS = 0.61181E-06 b/(keV*sr), CSC = 0.60211E-06 b/(keV*sr)
 theta =  72.0: CS = 0.61695E-06 b/(keV*sr), CSC = 0.60688E-06 b/(keV*sr)
 theta =  73.0: CS = 0.62382E-06 b/(keV*sr), CSC = 0.61334E-06 b/(keV*sr)
 theta =  74.0: CS = 0.63204E-06 b/(keV*sr), CSC = 0.62110E-06 b/(keV*sr)
 theta =  75.0: CS = 0.64123E-06 b/(keV*sr), CSC = 0.62979E-06 b/(keV*sr)
 theta =  76.0: CS = 0.65106E-06 b/(keV*sr), CSC = 0.63908E-06 b/(keV*sr)
 theta =  77.0: CS = 0.66122E-06 b/(keV*sr), CSC = 0.64866E-06 b/(keV*sr)
 theta =  78.0: CS = 0.67140E-06 b/(keV*sr), CSC = 0.65823E-06 b/(keV*sr)
 theta =  79.0: CS = 0.68133E-06 b/(keV*sr), CSC = 0.66752E-06 b/(keV*sr)
 theta =  80.0: CS = 0.69077E-06 b/(keV*sr), CSC = 0.67628E-06 b/(keV*sr)
 theta =  81.0: CS = 0.69947E-06 b/(keV*sr), CSC = 0.68430E-06 b/(keV*sr)
 theta =  82.0: CS = 0.70724E-06 b/(keV*sr), CSC = 0.69134E-06 b/(keV*sr)
 theta =  83.0: CS = 0.71387E-06 b/(keV*sr), CSC = 0.69724E-06 b/(keV*sr)
 theta =  84.0: CS = 0.71919E-06 b/(keV*sr), CSC = 0.70180E-06 b/(keV*sr)
 theta =  85.0: CS = 0.72305E-06 b/(keV*sr), CSC = 0.70489E-06 b/(keV*sr)
 theta =  86.0: CS = 0.72530E-06 b/(keV*sr), CSC = 0.70637E-06 b/(keV*sr)
 theta =  87.0: CS = 0.72582E-06 b/(keV*sr), CSC = 0.70610E-06 b/(keV*sr)
 theta =  88.0: CS = 0.72451E-06 b/(keV*sr), CSC = 0.70399E-06 b/(keV*sr)
 theta =  89.0: CS = 0.72126E-06 b/(keV*sr), CSC = 0.69995E-06 b/(keV*sr)
 theta =  90.0: CS = 0.71600E-06 b/(keV*sr), CSC = 0.69389E-06 b/(keV*sr)
 theta =  91.0: CS = 0.70866E-06 b/(keV*sr), CSC = 0.68576E-06 b/(keV*sr)
 theta =  92.0: CS = 0.69918E-06 b/(keV*sr), CSC = 0.67549E-06 b/(keV*sr)
 theta =  93.0: CS = 0.68752E-06 b/(keV*sr), CSC = 0.66304E-06 b/(keV*sr)
 theta =  94.0: CS = 0.67363E-06 b/(keV*sr), CSC = 0.64838E-06 b/(keV*sr)
 theta =  95.0: CS = 0.65749E-06 b/(keV*sr), CSC = 0.63148E-06 b/(keV*sr)
 theta =  96.0: CS = 0.63907E-06 b/(keV*sr), CSC = 0.61231E-06 b/(keV*sr)
 theta =  97.0: CS = 0.61835E-06 b/(keV*sr), CSC = 0.59086E-06 b/(keV*sr)
 theta =  98.0: CS = 0.59530E-06 b/(keV*sr), CSC = 0.56710E-06 b/(keV*sr)
 theta =  99.0: CS = 0.56989E-06 b/(keV*sr), CSC = 0.54100E-06 b/(keV*sr)
 theta = 100.0: CS = 0.54208E-06 b/(keV*sr), CSC = 0.51252E-06 b/(keV*sr)
 theta = 101.0: CS = 0.51181E-06 b/(keV*sr), CSC = 0.48161E-06 b/(keV*sr)
 theta = 102.0: CS = 0.47898E-06 b/(keV*sr), CSC = 0.44815E-06 b/(keV*sr)
 theta = 103.0: CS = 0.44342E-06 b/(keV*sr), CSC = 0.41199E-06 b/(keV*sr)
 theta = 104.0: CS = 0.40487E-06 b/(keV*sr), CSC = 0.37287E-06 b/(keV*sr)
 theta = 105.0: CS = 0.36289E-06 b/(keV*sr), CSC = 0.33034E-06 b/(keV*sr)
 theta = 106.0: CS = 0.31669E-06 b/(keV*sr), CSC = 0.28361E-06 b/(keV*sr)
 theta = 107.0: CS = 0.26472E-06 b/(keV*sr), CSC = 0.23114E-06 b/(keV*sr)
 theta = 108.0: CS = 0.20342E-06 b/(keV*sr), CSC = 0.16942E-06 b/(keV*sr)
 theta = 109.0: CS = 0.12048E-06 b/(keV*sr), CSC = 0.86689E-07 b/(keV*sr)
